src/commands/browseWebsite.ts

```diff
@@ -4,14 +4,18 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { IActionContext } from '@microsoft/vscode-azext-utils';
+import { functionFilter } from '../constants';
 import { ext } from '../extensionVariables';
 import { ResolvedFunctionAppResource } from '../tree/ResolvedFunctionAppResource';
 import { SlotTreeItem } from '../tree/SlotTreeItem';
 import { openUrl } from '../utils/openUrl';
 
 export async function browseWebsite(context: IActionContext, node?: SlotTreeItem): Promise<void> {
     if (!node) {
-        node = await ext.rgApi.tree.showTreeItemPicker<SlotTreeItem>(new RegExp(ResolvedFunctionAppResource.productionContextValue), context);
+        node = await ext.rgApi.pickAppResource<SlotTreeItem>(context, {
+            filter: functionFilter,
+            expectedChildContextValue: new RegExp(ResolvedFunctionAppResource.productionContextValue)
```

Comment:
Do you need the `expectedChildContextValue`?  Isn't the `productionContextValue` the `AppResource` itself?

Reply:
This unintentionally forces it to get resolved. And it doesn't behave weirdly. If we leave it out then we have to manually resolve it after calling `pickAppResource`.